chain/consensus/mir/state_manager.go

```diff
@@ -20,18 +21,32 @@ import (
 	lapi "github.com/filecoin-project/lotus/api"
 	"github.com/filecoin-project/lotus/api/v1api"
 	"github.com/filecoin-project/lotus/chain/actors/builtin"
+	"github.com/filecoin-project/lotus/chain/consensus/mir/validator"
 	"github.com/filecoin-project/lotus/chain/types"
 	ltypes "github.com/filecoin-project/lotus/chain/types"
 )
 
-var _ trantor.AppLogic = &StateManager{}
+var (
+	LatestCheckpointKey   = datastore.NewKey("mir/latest-check")
+	LatestCheckpointPbKey = datastore.NewKey("mir/latest-check-pb")
+)
 
 type Message []byte
 
 type Batch struct {
 	Messages []Message
 }
 
+type CtxCanceledWhileWaitingForBlockError struct {
+	Addr address.Address
+}
+
+func (e CtxCanceledWhileWaitingForBlockError) Error() string {
+	return fmt.Sprintf("validator %s context canceled while waiting for a snapshot", e.Addr)
+}
+
```

Comment:
What about moving these to a `types.go` file along with any other common type we may have?